### PR TITLE
Improved support for cambium cmm and added cmm4

### PR DESCRIPTION
**Pull request by pheinrichs**
- Added port discovery for cmm3 (not available for cmm4, not that i could find)
- Added discovery for cmm4
- Added Power port cmm4
- Removed Processor bug by disabling processor on device.

I am unable to test CMM4 and may need someone to do so or review for possible errors.

Fixes Issues #8732, #8731

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: tests/data/cmm4.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+    "os": {
+        "discovery": {
+            "devices": [
+                {
+                    "sysName": "",
+                    "sysObjectID": ".0.0.0.0.0.0.0.0.0.0",
```
Comment (laf) [inline tests/data/cmm4.json]: Is this genuinely the sysObjectID returned from SNMP?

**Comment by murrant** [comment]
I'm curious about the port pollers, looking at the json files a lot of data is missing.
cmm4 seems to not supply if-mib data like ifOperStatus and ifSpeed, are these available at the normal IF-MIB OIDs or just this device provides bad data?
cmm seem as though the port poller is totally disabled because it is just an empty file.

**Comment by pheinrichs** [comment]
@murrant I really can't confirm much on the cmm4. I don't have a device to confirm / test this on. 

After some more testing, the port implementation on cmm3 doesn't seem to be working quite right yet, Should merging an array to the port_stats var work? It seems as though the poller.php is attempting to insert into the ports_statistics table with a port_id of an empty string. Any idea what might be needed? 

If i purge the port_stats array entirely and just replace it with the newly created array that seems to work fine, just merging seems to cause issues.

**Comment by laf** [comment]
Don't think you can use array_merge() on multi dimensional arrays can you?

**Comment by murrant** [comment]
No, you would need to use something like array_replace_recursive()

**Comment by pheinrichs** [comment]
I think this PR is finally okay? After a long time coming ;)

**Pull request merged**